stackslib/src/clarity_vm/database/mod.rs

```
@@ -432,14 +433,21 @@ impl SortitionDBRef for SortitionHandleTx<'_> {
     }
 }
 
-impl SortitionDBRef for SortitionDBConn<'_> {
+impl SortitionDBRef for SortitionHandleConn<'_> {
     fn get_pox_start_cycle_info(
         &self,
         sortition_id: &SortitionId,
         parent_stacks_block_burn_ht: u64,
         cycle_index: u64,
     ) -> Result<Option<PoxStartCycleInfo>, ChainstateError> {
-        let mut handle = self.as_handle(sortition_id);
+        let readonly_marf = self
+            .index
+            .reopen_readonly()
+            .expect("BUG: failure trying to get a read-only interface into the sortition db.");
```

Can you just return an error here instead?